### PR TITLE
closingd: Correct whose turn it is when moving from initial negotiation to continuous negotiation.

### PR DESCRIPTION
**Pull request by ZmnSCPxj**
Fixes: #1361 

Reported-by: @nayuta-ueno

### CODE REVIEW

**Comment by rustyrussell** [comment]
Yech!  Valgrind didn't spot the uninitialized variable because it was on the stack :(

Fixing that in a separate patch should reveal the error, I think.  Let me try...

**Comment by ZmnSCPxj** [comment]
I have had a dozen different tries at the fix, they all fail for some reason; I will probably steal your code.

Also the uninitialized `allow_mistakes` field *was* caught in a Valgrind run of one of the various versions I was trying.  It is simply that the old version never managed to even enter that part of the code, since all our tests involve positive testing that both sides could come to an agreement.  Most of my various versions managed to diverge, entering the `feerange->allow_mistakes` check and triggering a Valgrind error.

**Comment by rustyrussell** [comment]
Most failures are subtle: we restrict the range to "previous value +/- 1", so you can't get stuck offering the same value over and over.  The one exception to this, of course, is the termination condition where both agree, so this always has to be checked before calling adjust_feerange().

In fact, adjust_feerange() could detect this case and return true, which might simplify our logic.

File: closingd/closing.c [hunk 1]
```diff
@@ -530,18 +530,12 @@ int main(int argc, char *argv[])
...
 	/* Now any extra rounds required. */
 	while (offer[LOCAL] != offer[REMOTE]) {
-		/* If they differ, adjust feerate. */
-		adjust_feerange(&cs, gossip_index, &channel_id, &feerange,
-				offer[whose_turn], whose_turn);
-
-		/* Now its the other side's turn. */
-		whose_turn = !whose_turn;
+		/* Still don't agree: adjust feerange based on previous offer */
+		adjust_feerange(&cs, gossip_index, &channel_id,
```
Comment (ZmnSCPxj) [inline closingd/closing.c]: It seems to me the below situation will erroneously pass?

0. `commitment_fee` = 1000
1. funder: 800
2. fundee: 200
3. funder: 900

`init_feerange` sets min 0, max 1000.  On entering the `while` loop we load the fundee fee into the feerange and set min 201, max 1000, but we do not constrain the funder side simllarly?

Comment (rustyrussell) [reply closingd/closing.c]: Agreed.  Fix coming...

Comment (rustyrussell) [reply closingd/closing.c]: Try those...

**Comment by cdecker** [comment]
ACK 7b66689

**Pull request merged**